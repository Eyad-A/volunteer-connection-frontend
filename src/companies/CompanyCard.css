.CompanyCard {
    max-height: 250px;
    margin-bottom: 15px;
    color: inherit;
    border-radius: 0;
    border: none;
    box-shadow: 0 1px 3px rgba(29, 29, 29, 0.096), 0 1px 3px rgba(0, 0, 0, 0.28);
}

.CompanyCard img {
    width: 50px;
}

.CompanyCard:hover {
    text-decoration: none;
    /* box-shadow: 0 3px 6px rgba(0,0,0,0.28), 0 3px 6px rgba(0,0,0,0.28); */
}