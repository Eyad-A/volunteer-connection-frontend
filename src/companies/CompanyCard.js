import React from "react";
import { Link } from "react-router-dom";
import "./CompanyCard.css";

/**
 * Company Card 
 * Shows information about a company 
 */

 function CompanyCard({ companyHandle, companyName, country, numEmployees, shortDescription, longDescription, websiteUrl, logoUrl, mainImageUrl, lookingFor }) {
  return (
    <div className="CompanyCard card">
      <div className="card-body">
        <h5>
          <Link to={`/companies/${companyHandle}`}>
          {companyName} 
          </Link>
          {logoUrl && <img src={logoUrl} alt={companyName} className="float-right ml-5" />}
        </h5>
        <p>{country}</p>
        <p>{numEmployees}</p>
        <p>{shortDescription}</p>
        <p>{longDescription}</p>
        <p>{lookingFor}</p>
        <p><a href={websiteUrl}></a></p>
        <p><img src={logoUrl} /></p>
        <p><img src={mainImageUrl} /></p>
      </div>
    </div>
  );
}

export default CompanyCard;