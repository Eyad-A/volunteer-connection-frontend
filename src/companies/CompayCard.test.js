import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

it("matches snapshot", function() {
    const { asFragment } = render(
        <MemoryRouter>
            <CompanyCard
                companyHandle="test"
                companyName="test company"
                state="Flordia"
                numEmployees='999'
                shortDescription="test"
                longDescription="test test"
                logoUrl="https://google.com"
                lookingFor="Designer"
            />
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});