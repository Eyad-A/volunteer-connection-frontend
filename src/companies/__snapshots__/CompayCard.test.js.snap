// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CompanyCard card"
  >
    <div
      class="card-body"
    >
      <h5>
        <a
          href="/companies/test"
        >
          test company
        </a>
        <img
          alt="test company"
          class="float-right ml-5"
          src="https://google.com"
        />
      </h5>
      <p>
        test
      </p>
      <p>
        Flordia
      </p>
      <p>
        999
      </p>
      <p>
        test
      </p>
      <p>
        Designer
      </p>
    </div>
  </div>
</DocumentFragment>
`;
